src/main/scala/mesosphere/util/monitor/HeartbeatMonitor.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.util.monitor
+
+import javax.inject.Inject
+
+import akka.actor._
+import akka.event.LoggingReceive
+import scala.concurrent.duration._
+
+// HeartbeatActor monitors the heartbeat of some process and executes handlers for various conditions.
```

Comment:
ack, unit test is already on the TODO list (see top of the PR).

we don't actually have a scheduler package. we have a bunch of scheduler-related stuff in mesosphere/marathon but i'd like to avoid polluting that more than necessary.

@jasongilanfarr you and @meichstedt are making similar points re: consolidating/moving the handler action implementations into this actor. I still don't follow the rationale for that, I'm wondering if I'm missing something obvious or part of a scala philosophy that's just not clicking for me. To me that would seem to take something that's mesos-scheduler-agnostic (a generic heartbeat monitor) and tightly couple it to our current scheduler/scheduler-driver callback implementation. I'd like to better understand the motivation for doing such. As it currently stands the heartbeat actor is a reusable component that does one generic thing. It's also testable outside of any scheduler abstraction. Why complicate it further?